src/lib/mina_base/parties.ml

```diff
@@ -454,7 +454,6 @@ let%test_module "Test" =
     let%test_unit "json roundtrip dummy" =
       [%test_eq: t] dummy (dummy |> Fd.to_json full |> Fd.of_json full)
 
-    let%test_unit "full circuit" = ()
-
-    (* TODO Fields_derivers_snapps.Test.Loop.run full dummy Party.Fields.names *)
+    let%test_unit "full circuit" =
+      Fields_derivers_snapps.Test.Loop.run full dummy |> ignore
```

Comment:
I believe this won’t actually run. It’s equivalent to saying “start running this code asynchronously and then ignore the promise” (don’t wait for it to run). We need a Thread_safe.block_on_async or to make the derivers not depend on deferred. See my WIP branch `feature/roundtrip-test`

Reply:
Based on how the code behaved when I ran it I'm sure this does run, and that it works. We should still add block_on_async or something similar that works with `Async_kernel`